windows compatibility: monorepo does not symlink

From @1337cookie https://github.com/keystonejs/keystone-5/pull/1463#issuecomment-519957463

@MadeByMike and I saw the same issue. It is not seen when using `admin` privillages but that  is definitely not advised and not a solution.

```
C:\Users\user\Documents\web\keystone-5>bolt
⚡️   bolt v0.23.6 (node v10.16.1)
�   [1/4] Validating project...
�   [2/4] Installing project dependencies...
$ C:\Users\user\AppData\Local\Yarn\Data\global\node_modules\bolt\node_modules\.bin\yarn config get user-agent yarn/1.17.3 npm/? node/v10.16.1 win32 x64
yarn install v1.17.3
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@1.2.7: The platform "win32" is incompatible with this module.
info "fsevents@1.2.7" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.4: The platform "win32" is incompatible with this module.
info "fsevents@1.2.4" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning "apollo-server-express > apollo-server-core > apollo-tracing@0.5.2" has incorrect peer dependency "graphql@0.10.x - 14.1.x".
warning "graphql-playground-react > react-codemirror@1.0.0" has incorrect peer dependency "react@>=15.5 <16".
warning "graphql-playground-react > react-codemirror@1.0.0" has incorrect peer dependency "react-dom@>=15.5 <16".
[4/4] Building fresh packages...
$ yarn build:build-field-types && preconstruct dev && node ./packages/build-field-types/bin.js dev  && yarn run --silent contributing-guide
yarn run v1.17.3
$ rimraf packages/build-field-types/lib && babel packages/build-field-types/src --out-dir packages/build-field-types/dist --ignore '**/__tests__/**/*','**/__mocks__/**/*'
Successfully compiled 42 files with Babel.
Done in 4.41s.
� info project is valid!
� error { [Error: EPERM: operation not permitted, symlink 'C:\Users\user\Documents\web\keystone-5\packages\arch\packages\alert\src\index.js' -> 'C:\Users\user\Documents\web\keystone-5\packages\arch\packages\alert\dist\alert.cjs.js.flow']
� error   errno: -4048,
� error   code: 'EPERM',
� error   syscall: 'symlink',
� error   path:
� error    'C:\\Users\\user\\Documents\\web\\keystone-5\\packages\\arch\\packages\\alert\\src\\index.js',
� error   dest:
� error    'C:\\Users\\user\\Documents\\web\\keystone-5\\packages\\arch\\packages\\alert\\dist\\alert.cjs.js.flow' }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
error Error: spawn C:/Users/user/AppData/Local/Yarn/Data/global/node_modules/bolt/node_modules/.bin/yarn ENOENT
error     at notFoundError (C:/Users/user/AppData/Local/Yarn/Data/global/node_modules/cross-spawn/lib/enoent.js:11:11)
error     at verifyENOENT (C:/Users/user/AppData/Local/Yarn/Data/global/node_modules/cross-spawn/lib/enoent.js:46:16)
error     at ChildProcess.cp.emit (C:/Users/user/AppData/Local/Yarn/Data/global/node_modules/cross-spawn/lib/enoent.js:33:19)
```

> Possibly [boltpkg/bolt#207](https://github.com/boltpkg/bolt/issues/207)

Yeah I also came across that one.

So I built the bolt package including the changes in this PR @ https://github.com/boltpkg/bolt/pull/215 I had to install and use the [WSL](https://docs.microsoft.com/en-us/windows/wsl/about) with ubuntu, node and yarn, to build it.

Using "yarn link" on the updated bolt directory in windows and "yarn link bolt" in the keystone directory, running "bolt", "bolt dev" and "bolt demo meetup dev" seems to work as expected, with the todo and meetup demo working properly in the browser.

<details>
<summary>Click here to see output from "bolt", "bolt dev" and "bolt demo meetup dev"</summary>

```
C:\Users\usernameremoved\Documents\web\keystone-5>bolt
⚡️   bolt v0.23.6 (node v10.16.1)
�   [1/4] Validating project...
�   [2/4] Installing project dependencies...
$ C:\Users\usernameremoved\Documents\web\bolt\node_modules\.bin\yarn config get user-agent yarn/1.12.3 npm/? node/v10.16.1 win32 x64
yarn install v1.12.3
[1/4] Resolving packages...
success Already up-to-date.
$ yarn build:build-field-types && preconstruct dev && node ./packages/build-field-types/bin.js dev  && yarn run --silent contributing-guide
yarn run v1.12.3
$ rimraf packages/build-field-types/lib && babel packages/build-field-types/src --out-dir packages/build-field-types/dist --ignore '**/__tests__/**/*','**/__mocks__/**/*'
Successfully compiled 42 files with Babel.
Done in 1.42s.
� info project is valid!
� success created links!
� info project is valid!
� success created links!
� Contributing to KeystoneJS?
� Read the full Contributing Guide (https://github.com/keystonejs/keystone-5/blob/master/CONTRIBUTING.md)
Done in 4.53s.
�   [3/4] Linking workspace dependencies...
�   [4/4] Linking workspace binaries...
�  success Installed and linked workspaces.
�   Done in 24.51s.

C:\Users\usernameremoved\Documents\web\keystone-5>bolt dev
⚡️   bolt v0.23.6 (node v10.16.1)
$ yarn demo todo dev
$ yarn --cwd ./demo-projects todo dev
$ yarn --cwd todo dev
$ cross-env NODE_ENV=development DISABLE_LOGGING=true keystone dev
i Command: keystone dev
√ Validated project entry file ./index.js
√ Keystone server listening on port 3000
√ Initialised Keystone instance
√ Connected to database
√ Keystone instance is ready �
� Keystone Admin UI:   http://localhost:3000/admin
� GraphQL Playground:  http://localhost:3000/admin/graphiql
� GraphQL API:         http://localhost:3000/admin/api

C:\Users\usernameremoved\Documents\web\keystone-5>bolt demo meetup dev
⚡️   bolt v0.23.6 (node v10.16.1)
$ yarn --cwd ./demo-projects meetup dev
$ yarn --cwd meetup dev
$ cross-env NODE_ENV=development DISABLE_LOGGING=true keystone dev
i Command: keystone dev
√ Validated project entry file ./index.js
√ Keystone server listening on port 3000
√ Initialised Keystone instance
- Connecting to database[ ready ] compiled successfully
\ Connecting to database[ wait ]  compiling ...
- Connecting to database[ ready ] compiled successfully
√ Connected to database
√ Keystone instance is ready �
� Keystone Admin UI:   http://localhost:3000/admin
� GraphQL Playground:  http://localhost:3000/admin/graphiql
� GraphQL API:         http://localhost:3000/admin/api
[ event ] build page: /
[ wait ]  compiling ...
[ ready ] compiled successfully
```
</details>

Running "bolt demo blog dev" however shows a new error which might not be related.

<details>
<summary>Click here to see "bolt demo blog dev" output</summary>

```C:\Users\usernameremoved\Documents\web\keystone-5>bolt demo blog dev
⚡️   bolt v0.23.6 (node v10.16.1)
$ yarn --cwd ./demo-projects blog dev
$ yarn --cwd blog dev
$ cross-env NODE_ENV=development DISABLE_LOGGING=true node -r dotenv-safe/config `yarn bin`/keystone dev
internal/modules/cjs/loader.js:638
    throw err;
    ^

Error: Cannot find module 'C:\Users\usernameremoved\Documents\web\keystone-5\demo-projects\blog\`yarn'
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:636:15)
    at Function.Module._load (internal/modules/cjs/loader.js:562:25)
    at Function.Module.runMain (internal/modules/cjs/loader.js:831:12)
    at startup (internal/bootstrap/node.js:283:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:622:3)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
error Error: spawn ../bolt/node_modules/.bin/yarn ENOENT
error     at notFoundError (C:/Users/usernameremoved/Documents/web/bolt/node_modules/cross-spawn/lib/enoent.js:11:11)
error     at verifyENOENT (C:/Users/usernameremoved/Documents/web/bolt/node_modules/cross-spawn/lib/enoent.js:46:16)
error     at ChildProcess.cp.emit (C:/Users/usernameremoved/Documents/web/bolt/node_modules/cross-spawn/lib/enoent.js:33:19)
```
</details>

Regardless I think the symlink issue can be solved by documenting the "developer mode" option in the docs for the demo projects and pushing for https://github.com/boltpkg/bolt/pull/215 to be merged/reassessed.

--- a/package.json
+++ b/package.json
@@ -97,7 +97,7 @@
     "mongodb": "^3.3.3",
     "p-is-promise": "^3.0.0",
     "pino-colada": "^1.4.5",
-    "preconstruct": "^0.1.3",
+    "preconstruct": "^0.3.2",
     "prettier": "^1.18.2",
     "prop-types": "^15.7.2",
     "pug": "2.0.4",

